src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/jpa/entity/PersonAddressEntity.kt

```diff
@@ -25,11 +26,26 @@ class PersonAddressEntity(
     referencedColumnName = "id",
     nullable = false,
   )
-  var person: PersonEntity,
+  var person: PersonEntity? = null,
 
   @Column
   val postcode: String? = null,
 
   @Version
   var version: Int = 0,
-)
+) {
+  companion object {
+    fun from(address: PersonAddress): PersonAddressEntity? {
+      if (!address.postcode.isNullOrEmpty()) {
```

Comment:
I find the ! at the start of the line tricky to spot - might be nicer to reverse this conditional or maybe use a `let`